backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/TextMultiValues.java

```diff
@@ -99,20 +99,25 @@ protected static class DocValuesTextMultiValues extends TextMultiValues {
 		@Override
 		public boolean advanceExact(int doc) throws IOException {
 			boolean found = values.advanceExact( doc );
-			nextOrd = found ? values.nextOrd() : NO_MORE_ORDS;
+			if ( found ) {
+				nextOrd = values.nextOrd();
+				docValueCount = values.docValueCount();
+				currentDocIndex = 0;
+			}
```

Comment:
Shouldn't there be an `else` here to reset these variables to `0`, just to be safe?